ZelBack/src/services/appsService.js

```diff
@@ -8016,6 +8039,38 @@ async function checkAndRequestApp(hash, txid, height, valueSat, i = 0) {
     } else {
       // update apphashes that we already have it stored
       await appHashHasMessage(hash);
+      // eslint-disable-next-line consistent-return
+      return true;
+    }
+  } catch (error) {
+    log.error(error);
+  }
+}
+
+/**
+ * To check and request an app. Handles fluxappregister type and fluxappupdate type.
+ * Verification of specification was already done except the price which is done here
+ * @param {object} apps array list with list of apps that are missing.
+ * @param {boolean} incoming If true the message will be asked to a incoming peer, if false to an outgoing peer.
+ * @param {number} i Defaults to value of 1.
+ * @returns {void} Return statement is only used here to interrupt the function and nothing is returned.
+ */
+async function checkAndRequestMultipleApps(apps, incoming = false, i = 1) {
+  try {
+    await requestAppsMessage(apps, incoming);
+    await serviceHelper.delay(30 * 1000);
+    const appsToRemove = [];
+    // eslint-disable-next-line no-restricted-syntax
+    for (const app of apps) {
+      // eslint-disable-next-line no-await-in-loop
+      const messageReceived = await checkAndRequestApp(app.hash, app.txid, app.height, app.valueSat, 2);
```

Comment:
We can be stuck in this loop for a long time

Reply:
No, as you can see we are calling the function with latest parameter 2, with latest parameter two we will just check if the message was received or not, nothing else is done.